api/controllers/comment.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+const express = require('express');
+const router = express.Router();
+
+const config = require('../config');
+const db = require('../models');
+const common = require('../common');
+
+
+/**
+ * @apiGroup Comment
+ * @api {get} /comment/project/:projectid  Get Comment List of Particular Project
+ * @apiHeader {String} authorization 
+ *                              A valid JWT token "Bearer: xxxxx"
+ * @apiSuccess {Object}            List of Comments 
+ */
+
+router.get('/project/:id', common.jwt(), (req, res, next) =>{
+    if(!req.params.id) return next("project id not defined");
+    let projectID = req.params.id;
+    let limit = req.query.limit || 100;
+    db.Projects.findById(projectID, async (err, project)=>{
+        if(err) return next(err);
+        if(!project) return res.status(404).end();
+        if(!common.ismember(req.user, project)) return res.status(401).end("you are not a member of this project");
+        const find = {"project": projectID, "removed" : false};
+        db.Comments.find(find).limit(+limit).exec((err, recs)=>{
+            // console.log(recs);
+            if(err) return next(err);
+            // console.log(recs);
+            db.Comments.countDocuments(find).exec((err, count)=>{
+                if(err) return next(err);
+                res.json(recs);
+            })
+        }); 
+    });
+});
+
+/**
+ * @apiGroup Comments
+ * @api {post} /comment/project/:id Post comment on a Project
+ * @apiParam {user_id} userID 
+ * @apiParam {String} comment
+ * @apiParam {project} projectID
+ * @apiSuccess {}            Comment registered 
+ */
+
+router.post('/', common.jwt(), function(req, res, next) {
+    if(!req.body.comment) next("comment string missing");
+    if(!req.body.project) next("project id missing");
+    let projectID = req.body.project;
+    db.Projects.findById(projectID, async (err, project)=>{
+        if(err) return next(err);
+        if(!project) return res.status(404).end();
+
+        let comment = new db.Comments(req.body);
```

Comment:
You are not resetting `project` field. A hacker can easily set the project field via req.body and make this comment show up on un-authorized project. 

Also, you are not checking for write access to the project. I think only the admins/members/guest users should be able to comment. For "public" project, any authenticated users should be able to comment.